### DISCUSSION TITLE
KANs are MLPs? MLP Equivalent

### DISCUSSION THREAD

**Discussion by hypnopump** [Blealtan/efficient-kan#31] [General]
[This note](https://colab.research.google.com/drive/1v3AHz5J3gk-vu4biESubJdOsUheycJNz#scrollTo=bzGAJGX6BsLW) shows equivalence of KAN to MLP in the piecewise linear approximation. I guess non-linearity of spline might help in some cases, but would be cool to have it as a baseline. [Here's the reddit discussion](https://www.reddit.com/r/MachineLearning/comments/1clcu5i/d_kolmogorovarnold_network_is_just_an_mlp/)

**Comment by minh-nguyenhoang** [comment]
Only if you are using b-splines of order 1

**Comment by Blealtan** [comment]
It's equivalent to activating the same hidden state with multiple activation functions and then use a wider linear transformation to shrink it back. Somewhat like Gated Linear Unit, but somewhat in reverse: linear transformation goes after the broadening activation.

**Comment by bilzard** [comment]
Hi. I also doubt fundamental difference between KANs and MLPs.

Suppose input dimension is equal to output dimension, KAN is mathematically equivalent to MLP with residual connections (see the rightmost picture in the below diagram).
Trainable non-linear activation function in KAN is different from fixed non-linear activation in MLP. However, I think they aren't fundamentally different.

Feel free to point out if I make some misunderstanding.

P.S. 
@Blealtan Thank you for releasing implementation of efficient-KAN. The above insight is deeply inspired by your implementation.

![Efficient-KAN drawio(1)](https://github.com/user-attachments/assets/42cc519a-8349-4897-9527-f0e06195b1ce)

**Reply by bilzard** [reply]
Sorry, the above statement is incorrect. f(x)+g(x) and g(f(x)+x)+f(x)+x are not generally equivalent.
